packages/frontend/src/components/SavedDashboards/AddTilesToDashboardModal.tsx

```diff
@@ -196,11 +228,60 @@ const AddTilesToDashboardModal: FC<AddTilesToDashboardModalProps> = ({
 
     const currentSpace = spaces?.find((s) => s.uuid === tile?.props.spaceUuid);
 
+    const isDashboardSelectItemDisabled = useCallback(
+        (dashboard: Dashboard) => {
+            const allDashboardTileTypes = Array.from(
+                new Set(dashboard.tiles.map((t) => t.type)),
+            );
+
+            switch (dashboardTileType) {
+                case DashboardTileTypes.SAVED_CHART:
+                case DashboardTileTypes.SQL_CHART:
+                    return allDashboardTileTypes.includes(
+                        DashboardTileTypes.SEMANTIC_VIEWER_CHART,
+                    );
+                case DashboardTileTypes.SEMANTIC_VIEWER_CHART:
+                    return (
+                        allDashboardTileTypes.includes(
+                            DashboardTileTypes.SAVED_CHART,
+                        ) ||
+                        allDashboardTileTypes.includes(
+                            DashboardTileTypes.SQL_CHART,
+                        )
+                    );
+                case DashboardTileTypes.LOOM:
+                case DashboardTileTypes.MARKDOWN:
+                    return false;
+                default:
+                    return assertUnreachable(
+                        dashboardTileType,
+                        `Unknown tile type: ${dashboardTileType}`,
+                    );
+            }
+        },
+        [dashboardTileType],
+    );
+
+    // ! Don't really like this here, it is suposed to be temporary until we make semantic layer a type of project connection
+    const dashboardsWithTiles = useAsync(() => {
+        return Promise.all(dashboards?.map((d) => getDashboard(d.uuid)) ?? []);
+    }, [dashboards]);
```

Comment:
The only solution without changing the backend too much is to request for all dashboard again.
Tried adding tiles to `DashboardBasicDetails`, but it would change the query a bit too much. We can still do this (I think it is a better approach) if we agree that this is not good at all

Reply:
nvm, returning we just need the tile types from the backend we don't actually need the whole tiles object, fixed here https://github.com/lightdash/lightdash/pull/11734/commits/5876a05873e26656085d4b81b24bd79109bd5d15